packages/expo-asset/src/Asset.js

```diff
@@ -193,28 +212,48 @@ export default class Asset {
     return asset;
   }
 
+  static fromUri(remoteUri) {
+    const hash = Buffer.from(remoteUri).toString('base64');
+    const pathArr = remoteUri.split('?')[0].split('/');
+    const filename = pathArr[pathArr.length - 1];
+    const [name, ext] = filename.split('.');
+    const arr = remoteUri.split('.');
+    const asset = new Asset({
+      name,
+      uri: remoteUri,
+      hash,
+      type: ext,
+      ignoreHashCheck: true,
+    });
+    return asset;
+  }
+
   async downloadAsync() {
     if (this.downloaded) {
       return;
     }
     if (this.downloading) {
-      await new Promise((resolve, reject) => this.downloadCallbacks.push({ resolve, reject }));
+      await new Promise((resolve, reject) =>
+        this.downloadCallbacks.push({ resolve, reject })
+      );
       return;
     }
     this.downloading = true;
     try {
-      const localUri = `${FS.cacheDirectory}ExponentAsset-${this.hash}.${this.type}`;
+      const localUri = `${FS.cacheDirectory}ExponentAsset-${this.hash}.${
+        this.type
+      }`;
       let exists, md5;
       ({ exists, md5 } = await FS.getInfoAsync(localUri, {
         cache: true,
         md5: true,
       }));
-      if (!exists || md5 !== this.hash) {
+      if (!exists || (!this.ignoreHashCheck && md5 !== this.hash)) {
```

Comment:
We usually want to check the hash and skip it just when we have no hash:

```suggestion
      if (!exists || (this.hash !== null && md5 !== this.hash)) {
```